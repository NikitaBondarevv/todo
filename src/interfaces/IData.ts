export interface IData {
  day?: string | number
  id?: string
  title?: string
  done?: boolean | undefined
  description?: string
  email?: string
  password?: string
  firstname?: string
  lastname?: string
}
