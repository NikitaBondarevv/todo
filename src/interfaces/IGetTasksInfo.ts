import { ReactNode } from 'react'

export interface IGetTasksInfo {
  text: string
  amount: ReactNode
  otherText?: string
  styles?: string
}
