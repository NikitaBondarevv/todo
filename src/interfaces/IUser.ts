export interface IUser {
  firstName: string
  lastname: string
  email: string
  password: string
  id?: string
  error?: string
}
