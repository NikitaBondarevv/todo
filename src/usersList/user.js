import { Component } from "react";

import styles from './styles.css'
import { getUsers } from "./getUsers";

export class User extends Component {
  state = {
    users: []
  }

  async getUsers() {
    this.originUsers = await getUsers()

    this.setState({ users: this.originUsers })
  }

  componentDidMount() {
    this.getUsers()
  }

  render() {
    const { users } = this.state

    return users.map((user, index) => <li key={index} className={styles.user}>{user.username}</li>)
  }
}
