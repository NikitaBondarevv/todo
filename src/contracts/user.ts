import { request } from './request'
import { IData } from 'interfaces/IData'

export const createUser = async (data: IData) => request.post('public/user', data)
export const updateUser = async (data: IData) => request.put('user', data)
