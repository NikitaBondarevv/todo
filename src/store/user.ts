import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'

import { IUser } from 'interfaces/IUser'

export interface UserState {
  data?: IUser
  isAuthenticated: boolean
}

const initialState: UserState = {
  isAuthenticated: false,
}

export const user = createSlice({
  name: 'user',
  initialState,
  reducers: {
    login: (state: UserState, action: PayloadAction<IUser>) => {
      state.data = action.payload
      state.isAuthenticated = true
    },
    
    logout: (state: UserState) => {
      delete state.data
      state.isAuthenticated = false
    },
  },
})

// Action creators are generated for each case reducer function
export const { login, logout } = user.actions

export default user.reducer
