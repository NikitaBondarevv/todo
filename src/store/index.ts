import { configureStore } from '@reduxjs/toolkit'
import { useDispatch, useSelector } from 'react-redux'
import type { TypedUseSelectorHook } from 'react-redux'

import user from './user'
import taskInfo from './taskInfo'

export const store = configureStore({
  reducer: {
    user: user,
    taskInfo: taskInfo
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
