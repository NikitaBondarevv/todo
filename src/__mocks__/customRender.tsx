import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'

import { store } from 'store'

export const customRender = (Cpmponent: React.ReactNode) => render(
  <BrowserRouter>
    <Provider store={store}>
      {Cpmponent}
    </Provider>
  </BrowserRouter>
)