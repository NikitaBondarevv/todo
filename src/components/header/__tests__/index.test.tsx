import { Header } from '..'
import { customRender } from '__mocks__/customRender'
import { login } from 'store/user'
import { store } from 'store'
import { IUser } from 'interfaces/IUser'

jest.mock('../authorizedUser', () => ({
  AuthorizedUser: () => <div data-test-id="test" />
}))

describe('<Header />', () => {
  test('should match snapshot', () => {
    const { asFragment } = customRender(<Header />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render <AuthorizedUser /> if "isAuthenticated" true', () => {
    const firstName = 'test name'
    store.dispatch(login({ firstName } as IUser))

    const { container } = customRender(<Header />)
    const authorizedUserElement = container.querySelector(`[data-test-id=test]`)

    expect(authorizedUserElement).toBeInTheDocument()
  })
})
