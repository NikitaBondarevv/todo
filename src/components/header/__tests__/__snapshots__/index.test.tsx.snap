// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<DocumentFragment>
  <header>
    <a
      href="/"
    >
      <img
        alt="logo"
        src="logo"
      />
    </a>
    <nav>
      <ul>
        <li>
          <a
            aria-current="page"
            class=" undefined"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class=" "
            href="/contacts"
          >
            Contacts
          </a>
        </li>
      </ul>
    </nav>
    <a
      class=" "
      href="/create"
    >
      Create User
    </a>
  </header>
</DocumentFragment>
`;
